docker-app/qfieldcloud/core/views/projects_views.py

```diff
@@ -46,6 +49,43 @@ def has_permission(self, request, view):
         return False
 
 
+class ProjectOrderingFilter(filters.OrderingFilter):
+    def filter_queryset(
+        self, request: HttpRequest, queryset: QuerySet, view: viewsets.ModelViewSet
+    ) -> QuerySet:
+        """
+        Filters a query set to sort it by ordering values, if any
+        Returns the ordered QuerySet
+        """
+        ordering = self.get_ordering(request, queryset, view)
+
+        if not ordering:
+            return queryset
+
+        new_ordering = []
+        for field in ordering:
+            descending = field.startswith("-")
+            clean_field = field.lstrip("-")
+
+            # owner's username is a FK
+            if field == "owner":
+                field = "owner__username"
```

Comment:
Can we try to make this more generic? What if the `field` is a `dict` that has a mapping `{"owner": "owner__username"}`? Or what if it s a string that has some kind of separator `owner:owner__username`?